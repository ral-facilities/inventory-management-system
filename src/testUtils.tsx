import { LocalizationProvider } from '@mui/x-date-pickers';
import { AdapterDateFns } from '@mui/x-date-pickers/AdapterDateFnsV3';
import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
import { RenderOptions, render } from '@testing-library/react';
import { enGB } from 'date-fns/locale/en-GB';
import React from 'react';
import {
  RouterProvider,
  createBrowserRouter,
  createMemoryRouter,
} from 'react-router-dom';
import { paths } from './App';
import {
  CatalogueCategory,
  CatalogueCategoryProperty,
  CatalogueItem,
  Item,
  Manufacturer,
} from './api/api.types';
import CatalogueCategoriesJSON from './mocks/CatalogueCategories.json';
import CatalogueItemsJSON from './mocks/CatalogueItems.json';
import ItemsJSON from './mocks/Items.json';
import ManufacturersJSON from './mocks/Manufacturers.json';

export const createTestQueryClient = (): QueryClient =>
  new QueryClient({
    defaultOptions: {
      queries: {
        retry: false,
        staleTime: 300000,
      },
    },
  });

interface ExtendedRenderOptions extends Omit<RenderOptions, 'queries'> {
  queryClient?: QueryClient;
}

function constructRouterProvider(
  ui: React.ReactNode,
  queryClient: QueryClient,
  urlPathKey?: keyof typeof paths,
  initialEntry?: string
) {
  const Root: React.FunctionComponent = () => {
    return (
      <LocalizationProvider adapterLocale={enGB} dateAdapter={AdapterDateFns}>
        <QueryClientProvider client={queryClient}>{ui}</QueryClientProvider>
      </LocalizationProvider>
    );
  };

  const router =
    initialEntry !== undefined && urlPathKey !== undefined
      ? createMemoryRouter([{ path: paths[urlPathKey], Component: Root }], {
          initialEntries: [initialEntry],
        })
      : createBrowserRouter([{ path: '*', Component: Root }]);

  return { router: router, provider: <RouterProvider router={router} /> };
}

function constructRouterProviderWrapper(
  queryClient: QueryClient,
  urlPathKey?: keyof typeof paths,
  initialEntry?: string
) {
  const wrapper = ({
    children,
  }: {
    children: React.ReactNode;
  }): JSX.Element => {
    return constructRouterProvider(
      children,
      queryClient,
      urlPathKey,
      initialEntry
    ).provider;
  };
  return wrapper;
}

export function renderComponentWithRouterProvider(
  ui: React.ReactElement,
  urlPathKey?: keyof typeof paths,
  initialEntry?: string,
  {
    // Automatically create a query client instance if no query client was passed in
    queryClient = createTestQueryClient(),
    ...renderOptions
  }: ExtendedRenderOptions = {}
) {
  const { router, provider } = constructRouterProvider(
    ui,
    queryClient,
    urlPathKey,
    initialEntry
  );
  return {
    queryClient,
    router,
    ...render(ui, {
      wrapper: () => provider,
      ...renderOptions,
    }),
  };
}

export const hooksWrapperWithProviders = (props?: {
  queryClient?: QueryClient;
  urlPathKey?: keyof typeof paths;
  initialEntry?: string;
}) => {
  const testQueryClient = props?.queryClient ?? createTestQueryClient();
  return constructRouterProviderWrapper(
    testQueryClient,
    props?.urlPathKey,
    props?.initialEntry
  );
};

export const getCatalogueItemsPropertiesById = (
  id: string
): CatalogueCategoryProperty[] => {
  const filteredCategories = CatalogueCategoriesJSON.filter(
    (catalogueCategory) => catalogueCategory.id === id
  );

  if (filteredCategories.length === 0) {
    return [];
  }

  const properties = filteredCategories[0].properties ?? [];
  return properties as CatalogueCategoryProperty[];
};

export const getCatalogueCategoryById = (
  id: string
): CatalogueCategory | undefined => {
  return (
    (CatalogueCategoriesJSON.find(
      (catalogueCategory) => catalogueCategory.id === id
    ) as CatalogueCategory) || undefined
  );
};

export const getCatalogueItemById = (id: string): CatalogueItem | undefined => {
  return (
    (CatalogueItemsJSON.find(
      (catalogueItem) => catalogueItem.id === id
    ) as CatalogueItem) || undefined
  );
};

export const catalogueItemData = (id: string): CatalogueItem[] => {
  return CatalogueItemsJSON.filter(
    (catalogueItem) => catalogueItem.catalogue_category_id === id
  );
};

export const getManufacturerById = (id: string): Manufacturer | undefined => {
  return (
    (ManufacturersJSON.find(
      (manufacturer) => manufacturer.id === id
    ) as Manufacturer) || undefined
  );
};

export const getItemsByCatalogueItemId = (catalogueItemId: string): Item[] => {
  return ItemsJSON.filter((item) => item.catalogue_item_id === catalogueItemId);
};

export const getItemById = (itemId: string): Item | undefined => {
  return ItemsJSON.find((item) => item.id === itemId);
};

export const getItemsBySystemId = (systemId: string): Item[] => {
  return ItemsJSON.filter((item) => item.system_id === systemId);
};

export const CREATED_MODIFIED_TIME_VALUES = {
  created_time: '2024-01-01T12:00:00.000+00:00',
  modified_time: '2024-01-02T13:10:10.000+00:00',
};
