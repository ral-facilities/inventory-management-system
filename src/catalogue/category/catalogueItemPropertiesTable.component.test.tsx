import userEvent, { UserEvent } from '@testing-library/user-event';
import { renderComponentWithRouterProvider } from '../../testUtils';
import PropertiesTable, {
  PropertiesTableProps,
} from './catalogueItemPropertiesTable.component';
import { screen, waitFor } from '@testing-library/react';

describe('CatalogeItemPropertiesTable', () => {
  let props: PropertiesTableProps;
  let user: UserEvent;

  const onChangeEditCatalogueItemField = vi.fn();
  const createView = () => {
    return renderComponentWithRouterProvider(<PropertiesTable {...props} />);
  };

  beforeEach(() => {
    props = {
      properties: [
        {
          name: 'Pumping Speed',
          type: 'number',
          //unit: 'liters per second',
          unit_id: '7',
          mandatory: true,
          allowed_values: {
            type: 'list',
            values: [
              {
                av_placement_id: 'av_placement_id_11',
                value: 300,
              },
              {
                av_placement_id: 'av_placement_id_12',
                value: 400,
              },
              {
                av_placement_id: 'av_placement_id_13',
                value: 500,
              },
            ],
          },
          cip_placement_id: 'cip_placement_id_14',
        },
        {
          name: 'Ultimate Pressure',
          type: 'number',
          //unit: 'millibar',
          unit_id: '8',
          mandatory: true,
          allowed_values: null,
          cip_placement_id: 'cip_placement_id_16',
        },
        {
          name: 'Axis',
          type: 'string',
          //unit: null,
          unit_id: null,
          mandatory: false,
          allowed_values: {
            type: 'list',
            values: [
              {
                av_placement_id: 'av_placement_id_17',
                value: 'y',
              },
              {
                av_placement_id: 'av_placement_id_18',
                value: 'x',
              },
              {
                av_placement_id: 'av_placement_id_19',
                value: 'z',
              },
            ],
          },
          cip_placement_id: 'cip_placement_id_20',
        },
      ],
      editingProperties: false,
      onChangeEditCatalogueItemField: onChangeEditCatalogueItemField,
      tableHeightPx: '600px',
    };

    user = userEvent.setup();
  });

  afterEach(() => {
    vi.clearAllMocks();
  });

  it('renders table correctly (not editing properties)', async () => {
    const { asFragment } = createView();
    await waitFor(() => {
      const propertyTable = screen.getByRole('table');
      expect(propertyTable).toBeInTheDocument();
    });
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders table correctly (editing properties)', async () => {
    props.editingProperties = true;

    const { asFragment } = createView();
    await waitFor(() => {
      const propertyTable = screen.getByRole('table');
      expect(propertyTable).toBeInTheDocument();
    });
    expect(asFragment()).toMatchSnapshot();
  });

  it('calls onChangeEditCatalogueItemField when selecting property', async () => {
    props.editingProperties = true;
    createView();

    await waitFor(() => {
      const propertyTable = screen.getByRole('table');
      expect(propertyTable).toBeInTheDocument();
    });

    const radioButton = screen.getAllByRole('radio', {
      name: 'Toggle select row',
    })[0];
    user.click(radioButton);

    expect(onChangeEditCatalogueItemField).toHaveBeenCalled();
  });
});
