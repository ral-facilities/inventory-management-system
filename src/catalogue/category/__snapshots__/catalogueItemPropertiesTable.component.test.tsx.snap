// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CatalogeItemPropertiesTable > renders table correctly (editing properties) 1`] = `
<DocumentFragment>
  <div
    style="width: 100%;"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation2 css-1atu56z-MuiPaper-root"
    >
      <div
        class="MuiTableContainer-root css-bkm888-MuiTableContainer-root"
        data-testid="properties-table-container"
      >
        <table
          class="MuiTable-root MuiTable-stickyHeader css-ts728e-MuiTable-root"
          style="--header-mrt_row_select-size: 60; --col-mrt_row_select-size: 60; --header-property_name-size: 250; --col-property_name-size: 250; --header-property_type-size: 200; --col-property_type-size: 200; --header-property_allowed_values-size: 300; --col-property_allowed_values-size: 300; --header-property_unit_id-size: 200; --col-property_unit_id-size: 200; --header-property_mandatory-size: 200; --col-property_mandatory-size: 200; --header-mrt_row_spacer-size: 40; --col-mrt_row_spacer-size: 40;"
        >
          <thead
            class="MuiTableHead-root css-q4cnup-MuiTableHead-root"
          >
            <tr
              class="MuiTableRow-root MuiTableRow-head css-ssspk8-MuiTableRow-root"
            />
          </thead>
          <tbody
            class="MuiTableBody-root css-19hv8cu-MuiTableBody-root"
          >
            <tr
              style="display: grid;"
            >
              <td
                colspan="7"
                style="display: grid;"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-4ohuce-MuiTypography-root"
                >
                  No records to display
                </p>
              </td>
            </tr>
          </tbody>
          <tfoot
            class="MuiTableFooter-root css-4ar9ic-MuiTableFooter-root"
          />
        </table>
      </div>
      <div
        class="MuiBox-root css-10gei56"
      >
        <div
          class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-entered css-v9zqoq-MuiCollapse-root"
          style="min-height: 0px;"
        >
          <div
            class="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
          >
            <div
              class="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
            >
              <span
                aria-busy="true"
                aria-label="Loading"
                class="MuiLinearProgress-root MuiLinearProgress-colorPrimary MuiLinearProgress-indeterminate css-qprkun-MuiLinearProgress-root"
                role="progressbar"
              >
                <span
                  class="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Indeterminate css-5ir5xx-MuiLinearProgress-bar1"
                />
                <span
                  class="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar2Indeterminate css-1r8wrcl-MuiLinearProgress-bar2"
                />
              </span>
            </div>
          </div>
        </div>
        <div
          class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-bz4dnt-MuiCollapse-root"
          style="min-height: 0px;"
        >
          <div
            class="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
          >
            <div
              class="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
            >
              <div
                class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiAlert-root MuiAlert-colorInfo MuiAlert-standardInfo MuiAlert-standard css-1pnnppi-MuiPaper-root-MuiAlert-root"
                role="alert"
              >
                <div
                  class="MuiAlert-message css-1pxa9xg-MuiAlert-message"
                >
                  <div
                    class="MuiStack-root css-5kul5x-MuiStack-root"
                  >
                    <div
                      class="MuiBox-root css-k008qs"
                    >
                       
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiBox-root css-1llu0od"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-13zw5gk-MuiTypography-root"
          >
            Returned 0 out of 3 Properties
          </p>
          <div
            class="MuiBox-root css-nokk8"
          >
            <div
              class="MuiTablePagination-root MuiBox-root css-8k4lth"
            >
              <div
                class="MuiBox-root css-exd1zr"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-animated MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-animated css-uqq6zz-MuiFormLabel-root-MuiInputLabel-root"
                  for="mrt-rows-per-page"
                >
                  Rows per page
                </label>
                <div
                  class="MuiInputBase-root MuiInput-root MuiInputBase-colorPrimary  css-1mmm5cp-MuiInputBase-root-MuiInput-root-MuiSelect-root"
                >
                  <div
                    aria-controls=":r1:"
                    aria-expanded="false"
                    aria-haspopup="listbox"
                    aria-label="Rows per page"
                    class="MuiSelect-select MuiSelect-standard MuiInputBase-input MuiInput-input css-1rxz5jq-MuiSelect-select-MuiInputBase-input-MuiInput-input"
                    role="combobox"
                    tabindex="0"
                  >
                    5
                  </div>
                  <input
                    aria-hidden="true"
                    aria-invalid="false"
                    class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                    id="mrt-rows-per-page"
                    tabindex="-1"
                    value="5"
                  />
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconStandard css-pqjvzy-MuiSvgIcon-root-MuiSelect-icon"
                    data-testid="ArrowDropDownIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M7 10l5 5 5-5z"
                    />
                  </svg>
                </div>
              </div>
              <nav
                aria-label="pagination navigation"
                class="MuiPagination-root MuiPagination-outlined css-1oj2twp-MuiPagination-root"
              >
                <ul
                  class="MuiPagination-ul css-wjh20t-MuiPagination-ul"
                >
                  <li>
                    <button
                      aria-label="Go to previous page"
                      class="MuiButtonBase-root Mui-disabled MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-outlined MuiPaginationItem-rounded MuiPaginationItem-colorSecondary MuiPaginationItem-outlinedSecondary Mui-disabled MuiPaginationItem-previousNext css-iqeo3-MuiButtonBase-root-MuiPaginationItem-root"
                      disabled=""
                      tabindex="-1"
                      type="button"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiPaginationItem-icon css-g2z002-MuiSvgIcon-root-MuiPaginationItem-icon"
                        data-testid="ChevronLeftIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M15.41 7.41 14 6l-6 6 6 6 1.41-1.41L10.83 12z"
                        />
                      </svg>
                    </button>
                  </li>
                  <li>
                    <button
                      aria-label="Go to next page"
                      class="MuiButtonBase-root Mui-disabled MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-outlined MuiPaginationItem-rounded MuiPaginationItem-colorSecondary MuiPaginationItem-outlinedSecondary Mui-disabled MuiPaginationItem-previousNext css-iqeo3-MuiButtonBase-root-MuiPaginationItem-root"
                      disabled=""
                      tabindex="-1"
                      type="button"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiPaginationItem-icon css-g2z002-MuiSvgIcon-root-MuiPaginationItem-icon"
                        data-testid="ChevronRightIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M10 6 8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                        />
                      </svg>
                    </button>
                  </li>
                </ul>
              </nav>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`CatalogeItemPropertiesTable > renders table correctly (not editing properties) 1`] = `
<DocumentFragment>
  <div
    style="width: 100%;"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation2 css-1atu56z-MuiPaper-root"
    >
      <div
        class="MuiTableContainer-root css-yaerxu-MuiTableContainer-root"
        data-testid="properties-table-container"
      >
        <table
          class="MuiTable-root MuiTable-stickyHeader css-ts728e-MuiTable-root"
          style="--header-property_name-size: 250; --col-property_name-size: 250; --header-property_type-size: 200; --col-property_type-size: 200; --header-property_allowed_values-size: 300; --col-property_allowed_values-size: 300; --header-property_unit_id-size: 200; --col-property_unit_id-size: 200; --header-property_mandatory-size: 200; --col-property_mandatory-size: 200; --header-mrt_row_spacer-size: 40; --col-mrt_row_spacer-size: 40;"
        >
          <thead
            class="MuiTableHead-root css-q4cnup-MuiTableHead-root"
          >
            <tr
              class="MuiTableRow-root MuiTableRow-head css-ssspk8-MuiTableRow-root"
            />
          </thead>
          <tbody
            class="MuiTableBody-root css-129p5vw-MuiTableBody-root"
          >
            <tr
              aria-label="Pumping Speed row"
              class="MuiTableRow-root MuiTableRow-root css-i1y727-MuiTableRow-root-MuiTableRow-root"
              data-index="0"
              role="row"
            />
            <tr
              aria-label="Ultimate Pressure row"
              class="MuiTableRow-root MuiTableRow-root css-i1y727-MuiTableRow-root-MuiTableRow-root"
              data-index="1"
              role="row"
            />
            <tr
              aria-label="Axis row"
              class="MuiTableRow-root MuiTableRow-root css-i1y727-MuiTableRow-root-MuiTableRow-root"
              data-index="2"
              role="row"
            />
          </tbody>
          <tfoot
            class="MuiTableFooter-root css-4ar9ic-MuiTableFooter-root"
          />
        </table>
      </div>
      <div
        class="MuiBox-root css-10gei56"
      >
        <div
          class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-bz4dnt-MuiCollapse-root"
          style="min-height: 0px;"
        >
          <div
            class="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
          >
            <div
              class="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
            >
              <div
                class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiAlert-root MuiAlert-colorInfo MuiAlert-standardInfo MuiAlert-standard css-3fnbcm-MuiPaper-root-MuiAlert-root"
                role="alert"
              >
                <div
                  class="MuiAlert-message css-1pxa9xg-MuiAlert-message"
                >
                  <div
                    class="MuiStack-root css-5kul5x-MuiStack-root"
                  >
                    <div
                      class="MuiBox-root css-k008qs"
                    >
                       
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiBox-root css-1llu0od"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-13zw5gk-MuiTypography-root"
          >
            Total Properties: 3
          </p>
          <div
            class="MuiBox-root css-nokk8"
          >
            <div
              class="MuiTablePagination-root MuiBox-root css-8k4lth"
            >
              <div
                class="MuiBox-root css-exd1zr"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-animated MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-animated css-uqq6zz-MuiFormLabel-root-MuiInputLabel-root"
                  for="mrt-rows-per-page"
                >
                  Rows per page
                </label>
                <div
                  class="MuiInputBase-root MuiInput-root MuiInputBase-colorPrimary  css-1mmm5cp-MuiInputBase-root-MuiInput-root-MuiSelect-root"
                >
                  <div
                    aria-controls=":r0:"
                    aria-expanded="false"
                    aria-haspopup="listbox"
                    aria-label="Rows per page"
                    class="MuiSelect-select MuiSelect-standard MuiInputBase-input MuiInput-input css-1rxz5jq-MuiSelect-select-MuiInputBase-input-MuiInput-input"
                    role="combobox"
                    tabindex="0"
                  >
                    5
                  </div>
                  <input
                    aria-hidden="true"
                    aria-invalid="false"
                    class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                    id="mrt-rows-per-page"
                    tabindex="-1"
                    value="5"
                  />
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconStandard css-pqjvzy-MuiSvgIcon-root-MuiSelect-icon"
                    data-testid="ArrowDropDownIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M7 10l5 5 5-5z"
                    />
                  </svg>
                </div>
              </div>
              <nav
                aria-label="pagination navigation"
                class="MuiPagination-root MuiPagination-outlined css-1oj2twp-MuiPagination-root"
              >
                <ul
                  class="MuiPagination-ul css-wjh20t-MuiPagination-ul"
                >
                  <li>
                    <button
                      aria-label="Go to previous page"
                      class="MuiButtonBase-root Mui-disabled MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-outlined MuiPaginationItem-rounded MuiPaginationItem-colorSecondary MuiPaginationItem-outlinedSecondary Mui-disabled MuiPaginationItem-previousNext css-iqeo3-MuiButtonBase-root-MuiPaginationItem-root"
                      disabled=""
                      tabindex="-1"
                      type="button"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiPaginationItem-icon css-g2z002-MuiSvgIcon-root-MuiPaginationItem-icon"
                        data-testid="ChevronLeftIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M15.41 7.41 14 6l-6 6 6 6 1.41-1.41L10.83 12z"
                        />
                      </svg>
                    </button>
                  </li>
                  <li>
                    <button
                      aria-current="true"
                      aria-label="page 1"
                      class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-outlined MuiPaginationItem-rounded MuiPaginationItem-colorSecondary MuiPaginationItem-outlinedSecondary Mui-selected MuiPaginationItem-page css-iqeo3-MuiButtonBase-root-MuiPaginationItem-root"
                      tabindex="0"
                      type="button"
                    >
                      1
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </button>
                  </li>
                  <li>
                    <button
                      aria-label="Go to next page"
                      class="MuiButtonBase-root Mui-disabled MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-outlined MuiPaginationItem-rounded MuiPaginationItem-colorSecondary MuiPaginationItem-outlinedSecondary Mui-disabled MuiPaginationItem-previousNext css-iqeo3-MuiButtonBase-root-MuiPaginationItem-root"
                      disabled=""
                      tabindex="-1"
                      type="button"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiPaginationItem-icon css-g2z002-MuiSvgIcon-root-MuiPaginationItem-icon"
                        data-testid="ChevronRightIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M10 6 8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                        />
                      </svg>
                    </button>
                  </li>
                </ul>
              </nav>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
